clients/ui/bff/internal/models/catalog_source_list.go

```diff
@@ -1,10 +1,17 @@
 package models
 
+type CatalogSourceStatus struct {
+	State   string  `json:"state,omitempty"`
+	Reason  *string `json:"reason,omitempty"`
+	Message *string `json:"message,omitempty"`
+}
+
 type CatalogSource struct {
-	Id      string   `json:"id"`
-	Name    string   `json:"name"`
-	Enabled *bool    `json:"enabled,omitempty"`
-	Labels  []string `json:"labels"`
+	Id      string               `json:"id"`
+	Name    string               `json:"name"`
+	Enabled *bool                `json:"enabled,omitempty"`
+	Labels  []string             `json:"labels"`
+	Status  *CatalogSourceStatus `json:"status,omitempty"`
```

Comment:
we will also have `Error` property- we need to add that here. Also need to update the mocks with this